File: lexilogicserver/src/main/java/org/amalgam/lexilogicserver/model/utilities/referenceobjects/PlayerGameDetail.java
package org.amalgam.lexilogicserver.model.utilities.referenceobjects;
import java.util.LinkedList;

public class PlayerGameDetail {
    private String username;
    private int points = 0;
    private LinkedList<String> words=new LinkedList<>();;
    private LinkedList<String> dupedWords=new LinkedList<>();
    private boolean ready = false;

    public PlayerGameDetail(String username) {
        this.username = username;
        words = new LinkedList<>();
        dupedWords = new LinkedList<>();
    }

    public LinkedList<String> getDupedWords() {
        return dupedWords;
    }

    public String getUsername() {
        return username;
    }

    public int getPoints() {
        int calculatedPoints = points;
        for(String word:dupedWords){
            if (listOfWordsContains(word)){
                calculatedPoints-= word.length();
            }
        }
        return calculatedPoints;
    }

    public void setPoints(int points){this.points = points;}

    public LinkedList<String> getWords() {
        return words;
    }

    public void addWord(String word){
        words.add(word);
    }

    public void addDupedWord(String word){
        dupedWords.add(word);
    }

    public boolean listOfWordsContains(String word){
        return words.contains(word);
    }

    public boolean listOfDupesContains(String word) { return dupedWords.contains(word);}

    public boolean isReady() {
        return ready;
    }

    public void setReady(boolean ready) {
        this.ready = ready;
    }
}

File: Client_Python/Utilities/Reference_Objects/PlayerGameDetail.py
from collections import deque


class PlayerGameDetail:
    def __init__(self, username):
        self._username = username
        self._points = 0
        self._words = deque()
        self._duped_words = deque()
        self._ready = False

    # Getter methods
    def get_duped_words(self):
        return self._duped_words

    def get_username(self):
        return self._username

    def get_points(self):
        return self._points

    def get_words(self):
        return self._words

    def is_ready(self):
        return self._ready

    # Setter methods
    def set_points(self, points):
        self._points = points

    def set_ready(self, ready):
        self._ready = ready

    # Functions
    def add_word(self, word):
        self._words.append(word)

    def add_duped_word(self, word):
        self._duped_words.append(word)

    def list_of_words_contains(self, word):
        return word in self._words
